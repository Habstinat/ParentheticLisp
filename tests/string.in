tests:
    - char
    - string
    - concatenation with builtin plus function

(   define
    (() ()())
    97
    (() (()()))
    [+ 7 [* 9 10]]
    (
        (() (()))
        ((()) ()()()()()()())
        (
            (() ()(()))
            ((()) ()()()()()()()()())
            ((()) ()()()()()()()()()())
        )
    )
)

97
(() (()()))

a
(
    (() (())(())())
    (() (()()))
)

aa
(
    (() (()))
    (
        (() (())(())())
        (() (()()))
    )
    (
        (() (())(())())
        (() (()()))
    )
)

(
    define
    (() ()())
    lst
    (() (())(())(()))

    (   
        cons
        (() ((()))())
        a
        (
            (() (())(())())
            (   
                (() (()))
                ((()) )
                (() (()()))
            )
        )

        (   
            cons
            (() ((()))())
            b
            (
                (() (())(())())
                (   
                    (() (()))
                    ((()) ())
                    (() (()()))
                )
            )

            (   
                cons
                (() ((()))())
                c
                (
                    (() (())(())())
                    (   
                        (() (()))
                        ((()) ()())
                        (() (()()))
                    )
                )

                (   
                    cons
                    (() ((()))())
                    d
                    (
                        (() (())(())())
                        (   
                            (() (()))
                            ((()) ()()())
                            (() (()()))
                        )
                    )
                    empty
                    (() ((())))
                    
                )
                
            )
            
        )
        
    )
)

lst
(() (())(())(()))

abcd
(   
    string
    (() (())()(()))

    lst
    (() (())(())(()))
)

abcdabcd
(
    (() (()))
    (   
        string
        (() (())()(()))

        lst
        (() (())(())(()))
    )
    (   
        string
        (() (())()(()))

        lst
        (() (())(())(()))
    )
)