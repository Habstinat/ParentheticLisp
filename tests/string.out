97.0 a aa ('a', ('b', ('c', ('d', ())))) abcd abcdabcd
